### PR TITLE
Fix the bug that `joint_attention_kwargs` is not passed to the FLUX's transformer attention processors

### PR DESCRIPTION
**Pull request by HorizonWind2004**
Issue link is below:

#9516

### CODE REVIEW

File: src/diffusers/models/transformers/transformer_flux.py
Comment (yiyixuxu) [inline src/diffusers/models/transformers/transformer_flux.py:165]: can you explain what additional argument you need to pass down to flux attention processor?

**Review by yiyixuxu — COMMENTED** [review]
thanks for the PR! I left a question

File: src/diffusers/models/transformers/transformer_flux.py
Comment (HorizonWind2004) [reply src/diffusers/models/transformers/transformer_flux.py:165]: Thank you for your recognition! 

In our work, I am trying to integrate box and mask into the FLUX model and implement layout control (similar to what has been done in many works on SD1.4). This requires modifying the attention processor. I believe that the architecture of FLUX and other transformers can also be used to develop better layout control algorithms, so I believe these modifications will contribute to future training-free experiments on FLUX.

Comment (yiyixuxu) [inline src/diffusers/models/transformers/transformer_flux.py:165]:
```suggestion
        joint_attention_kwargs=None,
```

File: src/diffusers/models/transformers/transformer_flux.py
Comment (yiyixuxu) [inline src/diffusers/models/transformers/transformer_flux.py:91]: should we pass this to `attn` too?

**Review by yiyixuxu — COMMENTED** [review]
thanks! let's support this :)

File: src/diffusers/models/transformers/transformer_flux.py
Comment (HorizonWind2004) [reply src/diffusers/models/transformers/transformer_flux.py:91]: Yes! I think it will be useful for other trial!

**Comment by HorizonWind2004** [comment]
I fixed a bug and now it is okay XD.

**Review by a-r-r-o-w — COMMENTED** [review]
thank you, looks good! could you run `make style` to fix the failing quality tests.

i have a question though: in [this](https://github.com/huggingface/diffusers/pull/9517#discussion_r1774753147) comment, you mention that your work uses this feature to modify something in, or pass additional arguments to, the attention processor. it is understandable why this would be useful, but we generally do not add features that are not testable or usable without a public implementation. is your work/method utilizing this change available for testing this? if it will be available in the near future, i think it might be best to postpone merging this PR after that

**Comment by a-r-r-o-w** [comment]
> thanks! let's support this :)

oh, i just saw yiyi's comment about being okay with supporting this. in that case, please disregard my question above. let's fix the quality tests and we should be okay to merge

**Comment by yiyixuxu** [comment]
@a-r-r-o-w 
I think, based on my undersanding, they just want to use a custom attention processor so they can do `pipe.transformer.set_attn_processor()`; they do not intend to add that to diffusers; 
but I think if we do not allow passing that kwargs, it won't work with a custom attention processor taking additional arguments. o

**Comment by HorizonWind2004** [comment]
> @a-r-r-o-w I think, based on my undersanding, they just want to use a custom attention processor so they can do `pipe.transformer.set_attn_processor()`; they do not intend to add that to diffusers; but I think if we do not allow passing that kwargs, it won't work with a custom attention processor taking additional arguments. o

Yes, that's it! QWQ

**Comment by HorizonWind2004** [comment]
@a-r-r-o-w 
I sincerely apologize for my late response. I have been busy with another project and forgot to check GitHub. I will proceed with fixing the issues that caused the quality tests to fail right away. TAT

**Comment by a-r-r-o-w** [comment]
Hi @HorizonWind2004, looks good to me! Could you run `make style` in the diffusers root folder and push the changes? Happy to merge after that once @yiyixuxu approves too

**Comment by HorizonWind2004** [comment]
> Hi @HorizonWind2004, looks good to me! Could you run `make style` in the diffusers root folder and push the changes? Happy to merge after that once @yiyixuxu approves too

Yes! Now it is okay.

**Review by yiyixuxu — APPROVED** [review]
thanks!

**Pull request merged**
